src/auto-evo/RunResults.cs

```diff
@@ -118,6 +118,27 @@ public void AddSplitResultForSpecies(Species species, Species splitSpecies, List
             results[splitSpecies].SplitFrom = species;
         }
 
+        public void KillSpeciesInPatch(Species species, Patch patch, bool refundMigrations = false)
+        {
+            AddPopulationResultForSpecies(species, patch, 0);
+
+            // We copy migration list to be able to modify it
+            foreach (var migration in results[species].SpreadToPatches.ToArray())
```

Comment:
For consistency with other code I'd use ToList here. If it turns out that ToArray is better in the usual case, we have a *ton* of code to convert.

Reply:
Actually, this is already a list, and so it seems to not have ToList() nor Copy() methods. Only ToArray() and CopyTo(var array). Figured out it'd be simpler to use an already existing method that might be faster.

Reply:
This is still not solved.